### ISSUE TITLE
Find all references in TS is off by 1 line in this specific case

### ISSUE THREAD

**Issue by alexdima** [microsoft/TypeScript#59800]
Appologies for the slow reaction in https://github.com/microsoft/TypeScript/issues/57888. Creating a new issue because I can't reopen the existing issue.

I can still reproduce this in latest vscode Insiders (tried 2 times and reproduced 2 times).

Steps:
* clone the vscode repository
* `git checkout c2d75edf8a7`
* open `src/vs/editor/common/viewModel/viewModelImpl.ts`
* reload window
* while the window is loading do in the terminal ``
* find all references on `ViewModel` (line 45)
* observe that the references in `codeEditorWidget.ts` are off by one line

https://github.com/user-attachments/assets/6b247ff2-1a28-4cd4-8f85-35cffef861b1

```
Version: 1.93.0-insider
Commit: ae45c9d4b0f71d53151edc6d18be09107903c229
Date: 2024-08-29T05:04:03.909Z
Electron: 30.4.0
ElectronBuildId: 10073054
Chromium: 124.0.6367.243
Node.js: 20.15.1
V8: 12.4.254.20-electron.0
OS: Darwin arm64 23.6.0
```

![Image](https://github.com/user-attachments/assets/aec3e657-70df-49ed-a403-839d5811b696)

**Comment by DanielRosenwasser** [comment]
It looks like you requested find-all-references while the full language service is not loaded. Please let me know if I've missed something.

**Comment by DanielRosenwasser** [comment]
Ah, I rewatched the video and see what you mean.

**Comment by alexdima** [comment]
> It looks like you requested find-all-references while the full language service is not loaded. Specifically, there's two processes - a partial server and a full server. The partial server tries its best to answer questions based on open files.

Yes, I tend to run a lot into the single-file server giving bad results for find references. I see how it's useful to give the illusion of speed, and it's a cool idea with it serving syntax based requests, etc. Do you know if there are some settings to opt out of it for the find references? I never invoke find references on a non-exported symbol, I always use F7 to loop through references if I want to go to usages in the current file. I only use find all references when I want to find usages in other files, so I would much prefer a "Loading..." experience where it eventually gives the correct results instead of the current local references which forces me to refresh or retrigger. I think I had a dicussion about this with @mjbvz or @jrieken in the past but they said that there is no API / event to refresh the result list once the real server kicks in, so that would turn into an api discussion.

**Comment by jrieken** [comment]
>  so I would much prefer a "Loading..." experience where it eventually gives the correct results instead of the current local references which forces me to refresh or retrigger.

I do agree and I think the current behaviour is wrong. Find all references should always yield the same result independent of where the extension is in its startup phase. 


Also correct that streaming is an ask but that is quite far out because UI stability will be an issue (when/how do you resort the result). Wrt to the API it might just be about returning an iterable over an array

**Comment by RyanCavanaugh** [comment]
Closing language service bugs related to the 6.0 implementation. For more information, see #62827
